src/browser/provider/host.js

```diff
@@ -0,0 +1,103 @@
+import { assignInWith } from 'lodash';
+import { getViewportSize } from 'testcafe-browser-natives';
+
+
+export default class BrowserProviderHost {
+    constructor (providerObject) {
+        assignInWith(this, providerObject, (hostProp, providerProp, propName) => {
+            var overloadedMethodName = '$' + propName;
```

Comment:
It's used to implement one-time-calling guard on `init` and `dispose`, see [here](https://github.com/DevExpress/testcafe/pull/573/files#diff-0a94afe44ac55486d1dfd5252a99ddeaR22). `init` checks initialization flags and calls providers `init`.